src/primitives/sinsemilla.rs

```diff
@@ -1,19 +1,37 @@
 //! The Sinsemilla hash function and commitment scheme.
 
-use group::Wnaf;
+use group::{Curve, Wnaf};
 use halo2::arithmetic::{CurveAffine, CurveExt};
 use pasta_curves::pallas;
 use subtle::CtOption;
 
-use crate::spec::{extract_p_bottom, i2lebsp};
-
 mod addition;
 use self::addition::IncompletePoint;
-
-mod constants;
 mod sinsemilla_s;
-pub use constants::*;
-pub(crate) use sinsemilla_s::*;
+pub use sinsemilla_s::SINSEMILLA_S;
+
+/// Number of bits of each message piece in $\mathsf{SinsemillaHashToPoint}$
+pub const K: usize = 10;
+
+/// $\frac{1}{2^K}$
+pub const INV_TWO_POW_K: [u8; 32] = [
+    1, 0, 192, 196, 160, 229, 70, 82, 221, 165, 74, 202, 85, 7, 62, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 240, 63,
+];
+
+/// The largest integer such that $2^c \leq (r_P - 1) / 2$, where $r_P$ is the order
+/// of Pallas.
+pub const C: usize = 253;
+
+// Sinsemilla Q generators
+
+/// SWU hash-to-curve personalization for Sinsemilla $Q$ generators.
+pub const Q_PERSONALIZATION: &str = "z.cash:SinsemillaQ";
+
+// Sinsemilla S generators
+
+/// SWU hash-to-curve personalization for Sinsemilla $S$ generators.
+pub const S_PERSONALIZATION: &str = "z.cash:SinsemillaS";
```

Comment:
Ah, now this file will end up in `halo2_gadgets`, so I presume this means that `src/constants/sinsemilla.rs` is intended to stay in `orchard`?